aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -224,8 +224,11 @@ public void addUserMetaData(Map<String, String> userMetaDataMap) {
 
         if (userMetaDataMap != null) {
             // Append each of the user-specified key-value pair to the user metadata for the connection
+            LOGGER.warn("Keynames 'SDK' and 'Version' are reserved and will be skipped");
             for (Map.Entry<String, String> metaData : userMetaDataMap.entrySet()) {
-                userMetadata.append("&" + metaData.getKey() + "=" + metaData.getValue());
+                if (!(metaData.getKey().equals("SDK") || metaData.getKey().equals("Version"))) {
+                    userMetadata.append("&" + metaData.getKey() + "=" + metaData.getValue());
+                }
```

Comment:
can you move the warn log message into else so we log only when those key names are used?

Reply:
Yes, Done